### ISSUE TITLE
C# should emit warning/error for ByValArray/ByValTStr without SizeConst

### ISSUE THREAD

**Issue by yizhang82** [dotnet/roslyn#18580]
**Version Used**: 
VS 2017. C# 2.0.0.61501

**Steps to Reproduce**:

1. Add following code
```csharp
    struct Foo
    {
        [MarshalAs(UnmanagedType.ByValArray)]
        int[] array;
    } 
```

**Expected Behavior**:

Warning that SizeConst is missing. C# defaults to 1 which rarely is the right thing to do. If we worry about compat we can add this on the highest warning level.

**Actual Behavior**:

No warning about SizeConst.

**Comment by jaredpar** [comment]
>  If we worry about compat we can add this on the highest warning level.

That isn't a sufficient mitigation for us.  There are too many customers out there who use the highest warning level and consider new warnings for existing code a back compat breaking change (aka adoption blocker).   

Given this code worked before, has use cases in the wild (based on the internal emai), taking this request is pretty risky.  I don't see how we could take this outside of warning waves feature support.

**Comment by tijoytom** [comment]
@jaredpar Is the "warning waves feature support" in ? If it's we would like to add this as a warning ie ByValArray usage without SizeConst

**Comment by jaredpar** [comment]
No. It's not in, nor is it presently scheduled for a release.
